src/main/java/seedu/address/model/reminder/ReminderList.java

```diff
@@ -54,23 +77,28 @@ public ReminderList delete(Reminder reminder) {
      * Finds a Reminder object to the current list.
      *
      * @param reminderDescription to search.
+     * @param reminderDate
      * @return the Reminder found. If there is no matching searches, returns null.
      */
-    public Reminder find(ReminderDescription reminderDescription) {
+    public Reminder find(ReminderDescription reminderDescription, ReminderDate reminderDate)
+            throws IllegalValueException {
         Reminder reminderFound = null;
         PriorityQueue<Reminder> reminderListCopy = new PriorityQueue<Reminder>(this.reminderPriorityQueue);
 
         while (!reminderListCopy.isEmpty()) {
             reminderFound = reminderListCopy.poll();
-            if (reminderFound.getDescription().equals(reminderDescription)) {
+            if (reminderFound.getDescription().equals(reminderDescription)
+                    && reminderFound.getReminderDate().equals(reminderDate)) {
                 break;
             }
         }
 
-        if (reminderFound.getDescription().equals(reminderDescription)) {
+        if (reminderFound.getDescription().equals(reminderDescription)
```

Comment:
The first time is to find the reminder, the second check is to check whether the reminderFound contains the incorrect one, since it is guaranteed to capture one reminder from the list (with the poll() function). But since I added the checkArgument() method, I believe it's safe to remove it :)